src/utils/asType.ts

```diff
@@ -0,0 +1,15 @@
+import { Class } from 'utility-types';
+
+export default function asType<T extends object, SubT extends T>(
```

Comment:
This was a weird one. Had to have `T` extend from object to get `constructor` to exist. Kinda makes sense but also not sure if this typing is correct

Reply:
I'd look at usage, but my hunch is this properly refactors to a much simpler type guard.